src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.OperationVisitor.cs

```diff
@@ -25,6 +26,32 @@ private sealed class OperationVisitor : GlobalFlowStateDataFlowOperationVisitor
                 _osPlatformType = osPlatformType;
             }
 
+            internal static bool TryParseGuardAttributes(ImmutableArray<AttributeData> attributes, ref GlobalFlowStateAnalysisValueSet value)
+            {
+                if (HasAnyGuardAttribute(attributes))
+                {
+                    using var infosBuilder = ArrayBuilder<PlatformMethodValue>.GetInstance();
+                    if (PlatformMethodValue.TryDecode(attributes, infosBuilder))
+                    {
+                        for (var i = 0; i < infosBuilder.Count; i++)
+                        {
+                            var newValue = GlobalFlowStateAnalysisValueSet.Create(infosBuilder[i]);
+                            value = i == 0 ? newValue : infosBuilder[i].Negated ? value.WithAdditionalAnalysisValues(newValue, false) :
```

Comment:
> infosBuilder[i].Negated ? value.WithAdditionalAnalysisValues(newValue, false)

Can you explain the reasoning behind this logic? This makes me feel uncomfortable, as `WithAdditionalAnalysisValues` was only designed to add to current global state when a new runtime platform guard is encountered. I am not able to figure out why we use `WithAdditionalAnalysisValues` for negated value but do the merge for non-negated value.

Reply:
`UnsupportedOSPlatformGuard` attributes parsed to `Negated` value when a member has a list of 
Unsupported guard attributes those need to be merged as AND logic, for example:
```cs
    [UnsupportedOSPlatformGuard("linux")]
    [UnsupportedOSPlatformGuard("Windows")]
    bool LinuxAndWindowsNotSupported { get; } = !OperatingSystem.IsLinux() && !OperatingSystem.IsWindows()
``` 
As you can see It should work same as checking condition `!OperatingSystem.IsLinux() && !OperatingSystem.IsWindows()`, for that emerging with `WithAdditionalAnalysisValues` worked well 

For Supported platform guards they need to be merged with OR, for `Merge` worked as needed
```cs
    [SupportedOSPlatformGuard("linux")]
    [SupportedOSPlatformGuard("Windows")]
    bool LinuxAndWindowsSupported { get; set; } = OperatingSystem.IsLinux() || OperatingSystem.IsWindows()
```